### ISSUE TITLE
Bad request response from Reddit while downloading MANY posts. Anyone else getting this? 

### ISSUE THREAD

**Issue by mapleweekend** [mapleweekend/easy-reddit-downloader#20]
I am getting this occasionally, is anyone else getting it? If so, can you post the log file? If not the log file, can you say the subreddit and the number of posts you were attempting to download?

**Comment by Ty-Grr** [comment]
attempted to download 1000 items from BestofRedditorUpdates but got caught at "Still downloading posts... (602/1000)"

only downloaded 230 items showing in the downloads folder.

attempting to run again

https://i.imgur.com/oLfkD4W.png

[2022 11 25 at 10-40-16.txt](https://github.com/mapleweekend/easy-reddit-downloader/files/10299764/2022.11.25.at.10-40-16.txt)

**Comment by mapleweekend** [comment]
Did rerunning it help? @Ty-Grr

**Comment by Ty-Grr** [comment]
Running it again brought same error
https://i.imgur.com/mtlPIFu.png


[2022 11 26 at 12-42-4.txt](https://github.com/mapleweekend/easy-reddit-downloader/files/10303472/2022.11.26.at.12-42-4.txt)

**Comment by mapleweekend** [comment]
Looking into it https://github.com/mapleweekend/easy-reddit-downloader/tree/20-bad-request-response-from-reddit-while-downloading-many-posts-anyone-else-getting-this

**Comment by mapleweekend** [comment]
@Ty-Grr I was able to fix this by moving to a new request module (Axios over HTTPRequest). I tested with 1000 posts on /r/BestofRedditorUpdates as well as /r/pics, /r/videos, and /r/news. Seems to be working well! 

One caveat, i had to adjust how I am considering the script "done". A few commits ago, i made it so it didn't count failed posts in the total but for now, i have to. 
So if you ask for 10 posts and it downloads 8 and 2 fail, it will stop the script. This is intentional and stopping that is a lot of work at the moment. Feel free to submit a new issue if you think it's important to ya! 


Let me know if you run into more issues with this new build.
